pkg/xray/reports.go

```diff
@@ -0,0 +1,802 @@
+							Optional:    true,
+							MaxItems:    1,
+							Description: "The release bundles to include into the report. Only one type of resource can be set per report.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"names": {
+										Type:        schema.TypeSet,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Set:         schema.HashString,
+										Optional:    true,
+										Description: "The list of release bundles names.",
+									},
+									"include_patterns": {
+										Type:        schema.TypeSet,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Set:         schema.HashString,
+										Optional:    true,
+										Description: "The list of include patterns",
+									},
+									"exclude_patterns": {
+										Type:        schema.TypeSet,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Set:         schema.HashString,
+										Optional:    true,
+										Description: "The list of exclude patterns",
+									},
+									"number_of_latest_versions": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										Default:      1,
+										ValidateFunc: validation.IntAtLeast(0),
+										Description:  "The number of latest release bundle versions to include to the report.",
+									},
+								},
+							},
+						},
+						"projects": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							MaxItems:    1,
+							Description: "The projects to include into the report. Only one type of resource can be set per report.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"names": {
+										Type:        schema.TypeSet,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Set:         schema.HashString,
+										Optional:    true,
+										Description: "The list of project names.",
+									},
+									"include_key_patterns": {
+										Type:        schema.TypeSet,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Set:         schema.HashString,
+										Optional:    true,
+										Description: "The list of include patterns.",
+									},
+									"number_of_latest_versions": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										Default:      1,
+										ValidateFunc: validation.IntAtLeast(0),
+										Description:  "The number of latest release bundle versions to include to the report.",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"filters": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "Advanced filters.",
+				Elem: &schema.Resource{
+					Schema: filtersSchema,
+				},
+			},
+		},
+	)
+}
+
+type Report struct {
+	ReportId   int        `json:"report_id,omitempty"`
+	Name       string     `json:"name"`
+	ProjectKey string     `json:"-"`
```

Comment:
We need to include 'project' support. If we don't want this as part of this PR, we need to add it as a fast-follow, and document the lack of support in the resource description.

Reply:
I'm merging the schemas using `getProjectKeySchema` already.